starknet_devnet/postman_wrapper.py

```diff
@@ -170,3 +177,82 @@ def load_mock_messaging_contract_in_l1(self, starknet, contract_address):
         self.l1_to_l2_message_filter = self.mock_starknet_messaging_contract.w3_contract.events.LogMessageToL2.createFilter(
             fromBlock="latest"
         )
+
+
+class Postman:
+    """
+    Postman class copied from starknet code base.
+    Modifications were made in _handle_l1_to_l2_messages function.
+    """
+
+    def __init__(
+        self,
+        mock_starknet_messaging_contract: EthContract,
+        starknet: Starknet,
+    ):
+        self.mock_starknet_messaging_contract = mock_starknet_messaging_contract
+        self.starknet = starknet
+        self.n_consumed_l2_to_l1_messages = 0
+
+        # Create a filter to collect LogMessageToL2 events.
+        w3_contract = self.mock_starknet_messaging_contract.w3_contract
+        self.message_to_l2_filter = w3_contract.events.LogMessageToL2.createFilter(
+            fromBlock="latest"
+        )
+
+    @classmethod
+    async def create(cls: Type[TPostman], eth_test_utils: EthTestUtils) -> TPostman:
+        """Execution of mock_starknet_messaging_contract."""
+        mock_starknet_messaging_contract = eth_test_utils.accounts[0].deploy(
+            MockStarknetMessaging, 0
+        )
+        starknet = await Starknet.empty()
+        return cls(
+            mock_starknet_messaging_contract=mock_starknet_messaging_contract,
+            starknet=starknet,
+        )
+
+    async def _handle_l1_to_l2_messages(self):
+        transactions_to_execute = []
+        for event in self.message_to_l2_filter.get_new_entries():
+            args = event.args
+            transaction = InternalL1Handler.create(
+                contract_address=args["toAddress"],
+                entry_point_selector=args["selector"],
+                calldata=[int(args["fromAddress"], 16), *args["payload"]],
+                nonce=args["nonce"],
+                chain_id=self.starknet.state.general_config.chain_id.value,
+            )
+            transactions_to_execute.append(transaction)
+            self.mock_starknet_messaging_contract.mockConsumeMessageToL2.transact(
```

Comment:
Does this `.transact(...)` change the blockchain state? If so, which one - L1 or L2?

Reply:
I'll check that.

Reply:
transact doesn't change state, it's just parsing something and throw exception

Reply:
Great! Resolving this discussion